### PR TITLE
Add guide on using multiple Python versions for local development

### PR DESCRIPTION
**Pull request by tswast**
This guide recommends `pyenv` and `pyenv-virtualenv`. I know there are many options, but this is what works for me.

### CODE REVIEW

File: MAC_SETUP.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    ```
+    pyenv virtualenv 3.6.4 python-docs-samples
+    ```
+
+    This creates a virtualenv folder within `~/.pyenv/versions/`.
+
+1.  Set the local Python version(s) with `pyenv local`
+
+    ```
+    pyenv local python-docs-samples 3.6.4 3.7.0 3.5.4 2.715
```
Comment (billyjacobson) [inline MAC_SETUP.md]: Could you include above this line what the 2 parameters are  like
#pyenv local [name of virtualenv] [list of python versions to use]
  pyenv local python-docs-samples 3.6.4 3.7.0 3.5.4 2.715

File: MAC_SETUP.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+1.  Install [pyenv](https://github.com/pyenv/pyenv).
+
+    I (tswast@) use [homebrew](https://brew.sh/) to install it.
+
+    ```
+    brew update
+    brew install pyenv
+    ```
+
+1.  Install the [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv)
```
[the remark this answers is not in the document]
Comment (tswast) [reply MAC_SETUP.md]: Added in https://github.com/GoogleCloudPlatform/python-docs-samples/pull/1621/commits/a739f4d112d342762704ed8fceeb866ee85380ee

File: MAC_SETUP.md (hunk 1 above)
Comment (tswast) [reply MAC_SETUP.md]: Done in https://github.com/GoogleCloudPlatform/python-docs-samples/pull/1621/commits/a739f4d112d342762704ed8fceeb866ee85380ee

File: MAC_SETUP.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+   If the Python version you want isn't listed, you may need to upgrade
+   your pyenv with homebrew.
+
+   ```
+   brew update
+   brew upgrade pyenv
+   ```
+
+1.  Compile the necessary Python versions with pyenv.
```
[the remark this answers is not in the document]
Comment (tswast) [reply MAC_SETUP.md]: Added in https://github.com/GoogleCloudPlatform/python-docs-samples/pull/1621/commits/327f506c674a370e56cbadb93e71baab2b87d2ff

File: MAC_SETUP.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    This creates a virtualenv folder within `~/.pyenv/versions/`.
+
+1.  Set the local Python version(s) with `pyenv local`
+
+    ```
+    # pyenv local [name of virtualenv] [list of python versions to use]
+    pyenv local python-docs-samples 3.6.4 3.7.0 3.5.4 2.7.15
+    ```
+
+1.  Now when you `cd` into the source directory or a subdirectory within it,
```
Comment (billyjacobson) [inline MAC_SETUP.md]: nit: s/Now/Now,

File: MAC_SETUP.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    pyenv will make your virtualenv the default Python. Since you specified
+    more than one version, it will also add binaries like `python36` and
+    `python27` to your PATH, which nox uses when picking Python interpreters.
+
+1.  Since we don't want to add the pyenv configuration to git, add
```
Comment (billyjacobson) [inline MAC_SETUP.md]: Rephrasing suggestion: Add .python-version to your global gitignore file, so it wont be committed into the repository.

Comment (tswast) [reply MAC_SETUP.md]: I like it! Done.

**Pull request merged**